src/expr/src/relation/func.rs

```diff
@@ -375,6 +377,22 @@ where
         })
 }
 
+fn array_agg<'a, I>(datums: I, temp_storage: &'a RowArena) -> Datum<'a>
+where
+    I: IntoIterator<Item = Datum<'a>>,
+{
+    // TODO(petrosagg): check if dimentions are correct
+    let datum = temp_storage.make_datum(|packer| {
+        let datums = datums.into_iter().collect_vec();
```

Comment:
I don't know anything about `array_agg` so this may be a silly question but should this filter out nulls the way jsonb_agg does?

Reply:
It shouldn't. But also, this is my draft PR mostly for CI runs so don't spend too many cycles reviewing it yet :)